bin/cli

```diff
@@ -84,34 +94,71 @@ const getChosenKitDependency = () => {
 
 (async () => {
   if (command === 'create') {
-    const installDirectory = getInstallLocation()
-    const kitDependency = getChosenKitDependency()
-
-    const copyFile = (fileName) => fs.copy(path.join(kitRoot, fileName), path.join(installDirectory, fileName))
-
-    await fs.ensureDir(installDirectory)
-    if ((await fs.readdir(installDirectory)).length > 0) {
-      console.error(`Directory ${installDirectory} is not empty, please specify an empty location.`)
-      process.exit(3)
+    // Install as a two-stage bootstrap process.
+    //
+    // In stage one we create an empty project folder and install
+    // govuk-prototype-kit and govuk-frontend, then bootstrap stage two from
+    // the newly installed package.
+    //
+    // In stage two we do the actual setup of the starter files.
+    //
+    // Doing it this way means we can be sure the version of the cli matches
+    // the version of the kit the user ends up with. Try to put as much logic
+    // as possible into stage two; stage one should ideally be able to install
+    // any future version of the kit.
+
+    const stage = additionalOptions[0] === '--' ? 'two' : 'one'
```

Comment:
Rather than having the stage one and two both called `create` can we call stage one `create` and stage two `generate-starter-files` (or something similar)?

Also, I realised we had a problem with the cli args and I've created a util for reading them.  Feel free to bring these changes into your branch if it helps (if you do I'll close my PR) https://github.com/alphagov/govuk-prototype-kit/pull/1622/files

Reply:
> Rather than having the stage one and two both called `create` can we call stage one create and stage two `generate-starter-files` (or something similar)?

Yeah we can do that. Do we want that stage two command to be documented and usable independently?